examples/java/src/main/java/org/apache/beam/examples/WindowedWordCount.java

```diff
@@ -211,19 +178,66 @@ public static void main(String[] args) throws IOException {
     PCollection<KV<String, Long>> wordCounts = windowedWords.apply(new WordCount.CountWords());
 
     /**
-     * Concept #5: Format the results for a BigQuery table, then write to BigQuery.
-     * The BigQuery output source supports both bounded and unbounded data.
+     * Concept #5: Customize the output format using windowing information
+     *
+     * <p>At this point, the data is organized by window. We're writing text files and and have no
+     * late data, so for simplicity we can use the window as the key and {@link GroupByKey} to get
+     * one output file per window. (if we had late data this key would not be unique)
+     *
+     * <p>To access the window in a {@link DoFn}, add a {@link BoundedWindow} parameter. This will
+     * be automatically detected and populated with the window for the current element.
      */
-    wordCounts.apply(ParDo.of(new FormatAsTableRowFn()))
-        .apply(BigQueryIO.Write
-          .to(getTableReference(options))
-          .withSchema(getSchema())
-          .withCreateDisposition(BigQueryIO.Write.CreateDisposition.CREATE_IF_NEEDED)
-          .withWriteDisposition(BigQueryIO.Write.WriteDisposition.WRITE_APPEND));
+    pipeline.getCoderRegistry().registerCoder(IntervalWindow.class, IntervalWindow.getCoder());
```

Comment:
I lost your other comment on the files/review screen, but I agree: This coder is pretty core; I've added it to the registry. We have ongoing talks about needing a standardized meta-format for windows (for custom windows the runner might benefit from being able to grab the `maxTimestamp`, even if the rest is a custom encoding)